### PR TITLE
Add notes to task details

### PR DESCRIPTION
**Pull request by jackra1n**
Simple field showing the count of notes and opening a dialog with the notes when clicked.

Do we also want to show notes for cases?
And should I also add the possibility to add notes or only read?

### CODE REVIEW

**Comment by alexsuter** [comment]
@jackra1n don't you want to try dynamic dialogs? we support this now.

**Comment by jackra1n** [comment]
> @jackra1n don't you want to try dynamic dialogs? we support this now.

Yes, @ivy-cst also told me that. I will change that 😉

**Comment by ivy-cst** [comment]
> @jackra1n don't you want to try dynamic dialogs? we support this now.

Already told him :+1:

**Comment by alexsuter** [comment]
image

**Comment by jackra1n** [comment]
image
image

File: dev-workflow-ui/src/ch/ivyteam/workflowui/NotesIvyDevWfBean.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+import javax.faces.bean.ManagedBean;
+import javax.faces.bean.ViewScoped;
+
+import ch.ivyteam.ivy.workflow.INote;
+import ch.ivyteam.workflowui.util.TaskUtil;
+
+@ManagedBean
+@ViewScoped
+public class NotesIvyDevWfBean {
```
Comment (alexsuter) [inline dev-workflow-ui/src/ch/ivyteam/workflowui/NotesIvyDevWfBean.java:13]: there is no need anymore to use "IvyDevWf". dev-workflow-ui is running a web-jar and if somebody installs it as iar, he/she could install it in it's own application.

File: dev-workflow-ui/src/ch/ivyteam/workflowui/tasks/TasksDetailsIvyDevWfBean.java [hunk 2]
```diff
@@ -204,6 +209,22 @@ public String getInfoBannerMessage() {
...
+            .modal(true)
+            .responsive(true)
+            .width("640")
+            .height("340")
+            .contentHeight("100%")
+            .contentWidth("100%")
+            .build();
+
+    Map<String, List<String>> params = new HashMap<>();
+    params.put("id", Arrays.asList(selectedTask.getUuid().toString()));
```
Comment (alexsuter) [inline dev-workflow-ui/src/ch/ivyteam/workflowui/tasks/TasksDetailsIvyDevWfBean.java:223]: List.of()

File: dev-workflow-ui/webContent/includes/template/frame-10.xhtml [hunk 3]
```diff
@@ -0,0 +1,31 @@
+<!DOCTYPE html>
```
Comment (alexsuter) [inline dev-workflow-ui/webContent/includes/template/frame-10.xhtml:1]: is this needed?

File: dev-workflow-ui/webContent/view/notesDialog.xhtml [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  xmlns:pe="http://primefaces.org/ui/extensions">
+<h:body>
+  <f:metadata>
+    <f:viewParam name="id" value="#{notesIvyDevWfBean.id}" />
+  </f:metadata>
+
+  <ui:composition template="../includes/template/frame-10.xhtml">
+    <ui:define name="title">Notes</ui:define>
+    <ui:define name="content">
+        <ui:repeat value="#{notesIvyDevWfBean.notes}" var="note">
```
Comment (alexsuter) [inline dev-workflow-ui/webContent/view/notesDialog.xhtml:13]: paging?
what happens if there are 1000 notes?
with big messages?

**Review by alexsuter — COMMENTED** [review]
Good first steps :+1:

File: dev-workflow-ui/webContent/includes/template/frame-10.xhtml (hunk 3 above)
Comment (alexsuter) [reply dev-workflow-ui/webContent/includes/template/frame-10.xhtml:1]: I mean the hole template

**Comment by jackra1n** [comment]
image

**Comment by jackra1n** [comment]
@ivy-lli because cst and als are on the beach 🏖️ 😎 you can check this when you have time

File: dev-workflow-ui/webContent/view/notesDialog.xhtml [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+body .ui-datatable .ui-datatable-data .notes-paginator-container {
+  border: 0;
+  padding: 0;
+}
+
+body .notes-paginator .ui-paginator-bottom {
+  border: 0;
+  padding: 0;
+}
+</style>
```
Comment (ivy-lli) [inline dev-workflow-ui/webContent/view/notesDialog.xhtml:29]: correct indent

File: dev-workflow-ui/webContent/view/notesDialog.xhtml [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+<h:body class="body-hd #{ivyFreyaTheme.mode}">
+  <ui:include src="/ivy-form/layouts/includes/progress-loader.xhtml" />
+  <div id="content" class="container frame">
+    <ui:insert name="content">
+      <h:form id="notesForm" styleClass="notes-form">
+        <p:dataTable id="notesPaginator" var="note" value="#{notesBean.notes}" rowKey="#{note.id}" rows="5"
+          paginator="true" paginatorPosition="bottom" styleClass="notes-paginator">
+          <p:column id="content" styleClass="notes-paginator-container">
+            <div class="mb-2 p-2 border-solid border-round border-1 border-black-alpha-10">
+              <div class="mb-2 text-lg">
```
Comment (ivy-lli) [inline dev-workflow-ui/webContent/view/notesDialog.xhtml:43]: why text-lg? I would leave it the normal size?

File: dev-workflow-ui/webContent/view/notesDialog.xhtml [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+</f:metadata>
+<h:body class="body-hd #{ivyFreyaTheme.mode}">
+  <ui:include src="/ivy-form/layouts/includes/progress-loader.xhtml" />
+  <div id="content" class="container frame">
+    <ui:insert name="content">
+      <h:form id="notesForm" styleClass="notes-form">
+        <p:dataTable id="notesPaginator" var="note" value="#{notesBean.notes}" rowKey="#{note.id}" rows="5"
+          paginator="true" paginatorPosition="bottom" styleClass="notes-paginator">
+          <p:column id="content" styleClass="notes-paginator-container">
+            <div class="mb-2 p-2 border-solid border-round border-1 border-black-alpha-10">
```
Comment (ivy-lli) [inline dev-workflow-ui/webContent/view/notesDialog.xhtml:42]: I would use the `--surface-d` color. mutch better in dark mode

File: dev-workflow-ui/webContent/view/notesDialog.xhtml
Comment (ivy-lli) [inline dev-workflow-ui/webContent/view/notesDialog.xhtml:1]: every xhtml file in the view folder is directly accessible. So I'm able to open this file without any permission or login check if I know the id of the task: https://nightly.demo.ivyteam.io/dev-workflow-ui_notes/faces/view/dev-workflow-ui/notesDialog.xhtml?id=f5da52e8-d54e-4dd8-84d4-dd99ce413fc7
I think this is not ideal, and we should add a login check.

**Review by ivy-lli — CHANGES_REQUESTED** [review]

File: dev-workflow-ui/webContent/view/notesDialog.xhtml [hunk 8]
```diff
@@ -35,12 +39,12 @@ body .notes-paginator .ui-paginator-bottom {
   <ui:include src="/ivy-form/layouts/includes/progress-loader.xhtml" />
   <div id="content" class="container frame">
     <ui:insert name="content">
-      <h:form id="notesForm" styleClass="notes-form">
+      <h:form id="notesForm" styleClass="notes-form" rendered="#{notesBean.hasUserPermission()}">
```
Comment (ivy-lli) [inline dev-workflow-ui/webContent/view/notesDialog.xhtml:42]: why not use an f:event and redirect to login page? e.g. as you do it on the task.xhtml?
`<f:event type="preRenderView" listener="#{tasksDetailsIvyDevWfBean.redirectIfCantResume}" />`

**Review by ivy-lli — APPROVED** [review]
Cool stuff. 
The case notes will be part of a next PR I assume?

**Comment by jackra1n** [comment]
> Cool stuff. The case notes will be part of a next PR I assume?

Yes I think I will do that as I think it should be pretty quick job, even tho Im not sure we were talking with Christian about case notes

**Comment by ivy-lli** [comment]
> > Cool stuff. The case notes will be part of a next PR I assume?
> 
> Yes I think I will do that as I think it should be pretty quick job, even tho Im not sure we were talking with Christian about case notes

https://github.com/axonivy/dev-workflow-ui/pull/383#issuecomment-2245141707

**Comment by jackra1n** [comment]
> > > Cool stuff. The case notes will be part of a next PR I assume?
> > 
> > 
> > Yes I think I will do that as I think it should be pretty quick job, even tho Im not sure we were talking with Christian about case notes
> 
> [#383 (comment)](https://github.com/axonivy/dev-workflow-ui/pull/383#issuecomment-2245141707)

ohh thanks haha 😅

**Pull request merged**